integration-tests/utils/integration-utils.js

```diff
@@ -171,8 +171,12 @@ const noSpinnersAfter = async (page, { action, debugMessage }) => {
   if (debugMessage) {
     console.log(`About to perform an action and wait for spinners. \n\tDebug message: ${debugMessage}`)
   }
-  const foundSpinner = page.waitForXPath('//*[@data-icon="loadingSpinner"]')
-  await Promise.all([foundSpinner, action()])
+  const foundSpinner = page.waitForXPath('//*[@data-icon="loadingSpinner"]', { timeout: 5000 })
+  await Promise.all([
+    // loadingSpinner is not always visible. Log but ignore error if loadingSpinner is not found
```

Comment:
This goes against the guarantee that the function is still asserting. Do we think that the guarantee isn't important anymore or is this just going to lead to another type of flaky failure (such as what led us to add this function in the first place)?

Reply:
Guarantee still is true for the 99.999% of time. @petesantos, can you help to explain this change? It's difficult for me to explain UI behavior with React.

Reply:
In case we don't reach an agreement, I'd revert back just this change and leave other changes in this PR.

Reply:
Oh, that's really interesting.

My first thought was: I wonder if there's something we can do to guarantee that spinners will be rendered for at least one frame. However, instead of effectively artificially slowing down the app, can we detect the effect of the action in this case and not care whether or not a spinner was rendered?